stacklok/codegate
# sqlite-vec vectorization database

Migrate to sqlite-vec

This PR migrates from weaviate to sqlite-vec.

I tried to keep the logic flow the same as before. Initial tests show prompt augmented correctly after matching a search.

<img width="694" alt="image" src="https://github.com/user-attachments/assets/db55c9d5-da37-42e2-84f9-802f1a93391a" />

## How to test

Load the packages into the database:

```bash
poetry run python scripts/import_packages.py --jsonl-dir=data   
```

Select a package from data and test in chat " is malicious-crates-dummy from crates safe to use?"

You should then see:

```bash
2024-12-24T17:13:26.3dZ [debug    ] Found matching packages in sqlite-vec database matched_packages=['malicious-crates-dummy (crates)'] module=codegate pathname=/Users/lhinds/repos/stacklok/codegate-repos/codegate/src/codegate/pipeline/codegate_context_retriever/codegate.py
```

 
Closes: #437
